draft-ietf-quic-recovery.md

```diff
@@ -509,6 +509,57 @@ frames containing them could be lost. In this case, the loss recovery algorithm
 may cause spurious retransmits, but the sender will continue making forward
 progress.
 
+## Tracking Sent Packets {#tracking-sent-packets}
+
+QUIC stores information about every packet sent. It's expected implementations
+will index this per-packet information by packet number and store the
+per-packet fields detailed below for loss recovery and congestion control.
+Additionally, implementations MUST ensure that any retransmittable frames
+being transmitted are tracked in case of loss.
+
+If a packet containing retransmittable frames is lost, the QUIC transport
+needs to recover from that loss, such as by retransmitting the data,
+sending an updated frame, or abandoning the frame.  For more information,
+see Section 13.2 of {{QUIC-TRANSPORT}}.
+
+Packets MUST be tracked until acknowledged or lost.  After a packet is lost,
+it SHOULD be tracked for an amount of time comparable to the maximum
+expected packet reordering, such as 1 RTT.  This allows detection of
+spurious retransmissions and MAY allow avoiding an extra retransmissions if
+the frames contained within the packet were retransmitted and lost again.
+
+Sent packets are tracked for each packet number space, and ACK
+processing only applies to a single space.
+
+### Sent Packet Fields {#sent-packets-fields}
+
+packet_number
+: The packet number of the sent packet.
+
+retransmittable:
+: A boolean that indicates whether a packet is retransmittable.
+  If true, it is expected that an acknowledgement will be received,
+  though the peer could delay sending the ACK frame containing it
+  by up to the MaxAckDelay.
+
+in_flight:
```

Comment:
Not a big deal, but I found it easier to just store "flightsize" which is set to zero if it doesn't count, instead of having in_flight and send_bytes. If you find that more compact and understandable, feel free to change it.